### PR TITLE
Feature/leapr

### PR DESCRIPTION
**Pull request by ameliajo**
Added pieces to allow for modern leapr to be integrated

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,457 +1,102 @@
...
-    target_link_libraries( njoy21_executable PUBLIC njoy21 )
-endif()
+add_library( njoy21_library INTERFACE
+    )
+target_include_directories( njoy21_library
+    INTERFACE ${CMAKE_CURRENT_BINARY_DIR}/src/
+    src/
+    )
+target_link_libraries( njoy21_library
+    INTERFACE ENDFtk
```
Comment (jlconlin) [inline CMakeLists.txt:65]: Probably will need to add leapr here.

File: cmake/develop_dependencies.cmake [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    GIT_TAG         origin/master
+    GIT_SHALLOW     TRUE
+    )
+
+FetchContent_Declare( utility
+    GIT_REPOSITORY  https://github.com/njoy/utility
+    GIT_TAG         origin/master
+    GIT_SHALLOW     TRUE
+    )
+
```
Comment (jlconlin) [inline cmake/develop_dependencies.cmake:43]: Will need to add `FetchContent_Declare( leapr...` here.

**Review by jlconlin — COMMENTED** [review]
So LEAPR has been added to the modernDirectory, but we still need to integrate the LEAPR code into here. Let's talk about how to do this.

File: cmake/develop_dependencies.cmake (hunk 2 above)
Comment (ameliajo) [reply cmake/develop_dependencies.cmake:49]: Thank you! That was my mistake to forget that. I updated the cmake files to have LEAPR information in them, but I'm getting a slight problem when I try to compile NJOY21, that it complains about LEAPR's dependence on ENDFtk. 

Are you able to compile this branch at all? At first I thought it was due to a discrepancy between NJOY21's GIT_TAG of ENDFtk and LEAPR's GIT_TAG for ENDFtk but resolving that didn't immediately seem to make a difference

Comment (jlconlin) [reply cmake/develop_dependencies.cmake:49]: I'm not able to compile this. After some investigation, I also can't compile the master branch of LEAPR, I think that is the problem. Are you able to compile LEAPR?

File: src/njoy21.hpp [hunk 3]
```diff
@@ -18,10 +18,7 @@
 #include "Log.hpp"
 #include "utility.hpp"
```
Comment (jlconlin) [inline src/njoy21.hpp:19]: I would have expected to see something like 

```c++
#include "leapr.hpp"
```
here. If we don't include leapr, then it won't get called. Can you explain why it's not there?

Comment (jlconlin) [reply src/njoy21.hpp:20]: I found it

Comment (ameliajo) [reply src/njoy21.hpp:20]: Oh heavens. Yes I think I'd added it in and then forgot to come back and formally merge this. I think that all the notes on this PR have already been addressed, should we close/delete it?

Comment (jlconlin) [reply src/njoy21.hpp:20]: I'm still doing a little investigating.

**Pull request merged**